service/worker/scheduler/workflow_test.go

```diff
@@ -1669,6 +1670,71 @@ func (s *workflowSuite) TestUpdateNotRetroactive() {
 	)
 }
 
+// Tests that an update between a nominal time and jittered time for a start, that doesn't
+// modify that start, will still start it.
+func (s *workflowSuite) TestUpdateBetweenNominalAndJitter() {
+	// TODO: remove once default version is UpdateFromPrevious
+	prevTweakables := currentTweakablePolicies
+	currentTweakablePolicies.Version = UpdateFromPrevious
+	defer func() { currentTweakablePolicies = prevTweakables }()
+
+	spec := &schedpb.ScheduleSpec{
+		Interval: []*schedpb.IntervalSpec{{
+			Interval: durationpb.New(1 * time.Hour),
+		}},
+		Jitter: durationpb.New(1 * time.Hour),
```

Comment:
non-blocking question: where does this translate into the seemingly random yet reproducible jitter times below?

Reply:
the jitter seed (nsid + schedule id) is passed from the workflow to getNextTime on the CompiledSpec, which hashes that plus the time itself to pick a jitter